src/i18n/locales/index.ts

```
@@ -0,0 +1,33 @@
+/**
+ * Generates a messages schema by merging locale data for each language.
+ *
+ * @param locales - A record where the key is the path to the locale file and the value is the module import interface.
+ * @returns A messages schema object where each key is a language code and the value is the merged locale data for that language.
+ */
+const getLocales = (locales: Record<string, ModuleImportInterface>) => {
+  return languages.reduce((messages, lang) => {
+    messages[lang] = Object.keys(locales)
+      .filter(path => path.includes(`/${lang}.json`))
+      .reduce((localisedStrings, path) => {
+        return { ...localisedStrings, ...locales[path].default };
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)